disk/test/com/treode/disk/edit/DiskIOSpec.scala

```diff
@@ -18,62 +18,150 @@ package com.treode.disk.edit
 
 import com.treode.async.io.stubs.StubFile
 import com.treode.async.stubs.StubScheduler
-import com.treode.async.stubs.implicits._
 import com.treode.disk.DiskTestConfig
 import org.scalatest.FlatSpec
+import com.treode.pickle.Picklers._
+import com.treode.disk.PageDescriptor
+import com.treode.disk.Position
+import com.treode.async.{Async, Callback, Fiber, Scheduler }, Async.async
+import com.treode.disk.PickledPage
+import com.treode.async.stubs.implicits._
+
 
 class DiskIOSpec extends FlatSpec {
 
   implicit val config = DiskTestConfig()
-
-  "The PageReader" should "be able to read the string the PageWriter wrote" in {
+   "The PageReader" should "be able to read the tuple the PageWriter wrote" in {
     implicit val scheduler = StubScheduler.random()
-    val a = "this is a string"
+    val stringPickler = 
+        wrap(string,string)
+        .build(x => (x._1, x._2) )
+        .inspect( x=> (x._1, x._2) )
+
+    val a = "this is "
+    val b = "a string!"
     val readPos = 0
     val f = StubFile (1 << 20, 0)
-
-    val dw = new PageWriter (f)
+    val dsp = new PageDispatcher
+    val dw = new PageWriter (dsp, f)
     val dr = new PageReader (f)
-    val (pos, len) = dw.write (a) .expectPass()
-    val readString = dr.readString (readPos, len) .expectPass()
```

Comment:
You can save yourself from introducing temp variables.

    assert (a == dr.readSring (readPos, len) .expectPass())

Or

    dr.readString (readPos, len) .expectPass (a)

Also, in Scala we use `==` rather than `.equals`. In Java `==` is reference equality, but in Scala it's object equality.